document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -1190,6 +1215,25 @@ private void handleNestedField(
     }
   }
 
+  private static JsonNode mapValueToJsonNode(int columnType, String columnValue) {
+    switch (columnType) {
+      case VARCHAR:
+        return TextNode.valueOf(columnValue);
+      case BIGINT:
+        return BigIntegerNode.valueOf(BigInteger.valueOf(Long.valueOf(columnValue)));
+      case INTEGER:
+        return IntNode.valueOf(Integer.valueOf(columnValue));
+      case BOOLEAN:
+        return BooleanNode.valueOf(Boolean.valueOf(columnValue));
+      case DOUBLE:
+        return DoubleNode.valueOf(Double.valueOf(columnValue));
+      case FLOAT:
+        return FloatNode.valueOf(Float.valueOf(columnValue));
+      default:
+        return null;
```

Comment:
Can we prefer returning Optional instead of null?
That way, the callers would be forced to check for the presence.